lib/spark/types.ex

```diff
@@ -15,18 +15,26 @@ defmodule Spark.Types do
     doc_type({:or, [{:fun, arity}, {:spark_behaviour, mod}]})
   end
 
-  def doc_type({:custom, _, _, _}), do: "`any`"
-  def doc_type(:any), do: "`any`"
-  def doc_type(:keyword_list), do: "Keyword.t"
+  def doc_type({:custom, _, _, _}), do: "any"
+  def doc_type(:any), do: "any"
 
-  def doc_type({:keyword_list, schema}) when is_list(schema) do
-    options = Enum.map(schema, fn {key, value} -> {key, value[:type]} end)
-    "#{doc_type({:or, options})}"
-  end
+  def doc_type(:keyword_list), do: "keyword"
+  def doc_type(:non_empty_keyword_list), do: "keyword"
+
+  def doc_type({keyword_type, [*: value]})
+      when keyword_type in [:keyword_list, :non_empty_keyword_list] and is_list(value),
+      do: "keyword(#{doc_type(value[:type])})"
 
-  def doc_type({:keyword_list, value_type}), do: "Keyword.t(#{doc_type(value_type)})"
+  def doc_type({keyword_type, schema})
+      when keyword_type in [:keyword_list, :non_empty_keyword_list] and is_list(schema) do
+    options =
+      Enum.map_join(schema, ", ", fn {key, value} ->
+        "#{key}: #{doc_type(value[:type])}"
```

Comment:
I merged this, but it may also end up making huge doc strings...will need to see this in practice, might need to walk it back.

Reply:
I made it so because previous `doc_type` for `keyword_list` with `schema` had such intention.

Also since it was throwing errors before it means nobody was using such functionality before, so it is their choice to make it more detailed or list it as a simple `keyword_list`.